File: app/src/main/java/com/example/spring/RegisterFeatureRequest.java
package com.example.spring;

/**
 * Request DTO for command RegisterFeature.
 */
public record RegisterFeatureRequest(
    boolean active,
    @jakarta.validation.constraints.NotBlank String defaultUnit,
    @jakarta.validation.constraints.NotBlank String displayName,
    @jakarta.validation.constraints.NotBlank String featureKey,
    @jakarta.validation.constraints.NotBlank String tenantId,
    @jakarta.validation.constraints.NotBlank String valueType
) {}

File: app-python/entitlements_app/requests/register_feature_request.py
"""Pydantic request model for command `RegisterFeature`."""

from pydantic import BaseModel


class RegisterFeatureRequest(BaseModel):
    active: bool
    defaultUnit: str
    displayName: str
    featureKey: str
    tenantId: str
    valueType: str

    @property
    def default_unit(self) -> str:
        return self.defaultUnit

    @property
    def display_name(self) -> str:
        return self.displayName

    @property
    def feature_key(self) -> str:
        return self.featureKey

    @property
    def tenant_id(self) -> str:
        return self.tenantId

    @property
    def value_type(self) -> str:
        return self.valueType
